mm2src/mm2_main/src/rpc/dispatcher/dispatcher.rs

```diff
@@ -200,3 +205,31 @@ async fn dispatcher_v2(request: MmRpcRequest, ctx: MmArc) -> DispatcherResult<Re
         _ => MmError::err(DispatcherError::NoSuchMethod),
     }
 }
+
+/// `gui_storage_rpc` dispatcher.
+///
+/// # Note
+///
+/// `gui_storage_method` is a method name with the `gui_storage::` prefix removed.
+async fn gui_storage_dispatcher(
+    request: MmRpcRequest,
+    ctx: MmArc,
+    gui_storage_method: String,
```

Comment:
This can be better with `&str` type. This value is getting cast into `String` from `&str` in `dispatcher_v2` function and then in this function getting cast back into `&str`. We can leave it as it is and it should be just fine :)

Reply:
I tried to do it already, but got this compile error:
```
error[E0505]: cannot move out of `request` because it is borrowed
   --> mm2src/mm2_main/src/rpc/dispatcher/dispatcher.rs:125:39
    |
124 |     if let Some(gui_storage_method) = request.method.strip_prefix("gui_storage::") {
    |                                       -------------------------------------------- borrow of `request.method` occurs here
125 |         return gui_storage_dispatcher(request, ctx, gui_storage_method).await;
    |                                       ^^^^^^^       ------------------ borrow later used here
    |                                       |
    |                                       move out of `request` occurs here
```
It seems to be impossible since I move `request` to `gui_storage_dispatcher`

Reply:
What about the following diff?

```diff
diff --git a/mm2src/mm2_main/src/rpc/dispatcher/dispatcher.rs b/mm2src/mm2_main/src/rpc/dispatcher/dispatcher.rs
index 72b686b4c..fbcdd1960 100644
--- a/mm2src/mm2_main/src/rpc/dispatcher/dispatcher.rs
+++ b/mm2src/mm2_main/src/rpc/dispatcher/dispatcher.rs
@@ -123,7 +123,7 @@ async fn auth(request: &MmRpcRequest, ctx: &MmArc, client: &SocketAddr) -> Dispa
 async fn dispatcher_v2(request: MmRpcRequest, ctx: MmArc) -> DispatcherResult<Response<Vec<u8>>> {
     if let Some(gui_storage_method) = request.method.strip_prefix("gui_storage::") {
         let gui_storage_method = gui_storage_method.to_owned();
-        return gui_storage_dispatcher(request, ctx, gui_storage_method).await;
+        return gui_storage_dispatcher(request, ctx, &gui_storage_method).await;
     }

     match request.method.as_str() {
@@ -214,11 +214,11 @@ async fn dispatcher_v2(request: MmRpcRequest, ctx: MmArc) -> DispatcherResult<Re
 async fn gui_storage_dispatcher(
     request: MmRpcRequest,
     ctx: MmArc,
-    gui_storage_method: String,
+    gui_storage_method: &str,
 ) -> DispatcherResult<Response<Vec<u8>>> {
     use mm2_gui_storage::rpc_commands as gui_storage_rpc;

-    match gui_storage_method.as_str() {
+    match gui_storage_method {
         "activate_coins" => handle_mmrpc(ctx, request, gui_storage_rpc::activate_coins).await,
         "add_account" => handle_mmrpc(ctx, request, gui_storage_rpc::add_account).await,
         "deactivate_coins" => handle_mmrpc(ctx, request, gui_storage_rpc::deactivate_coins).await,
```

This way we don't need to re-cast `gui_storage_method` back to the initial type.

Reply:
@sergeyboyko0791 Did you consider the above suggestion from Onur?

Reply:
Not yet, will do right now